### PR TITLE
[Core Cherry-pick] Increase open ldb files

### PR DESCRIPTION
**Pull request by ptschip**
This is something I've wanted to investigate for a while and looks like someone did a really good job of studying and testing this out.  See : https://github.com/bitcoin/bitcoin/pull/12495

It seems to have a very good improvement on performance although probably for more marginal systems running on spinning disk and that have lower RAM but still, it's a simple fix with large impact.

Also pulled in quite a few developer doc updates that came along with the cherry-pick.

### CODE REVIEW

File: doc/developer-notes.md [hunk 1]
```diff
@@ -470,3 +470,251 @@ GUI
...
+  - Upstream at https://github.com/jgarzik/univalue ; report important PRs to Core to avoid delay.
+
+Upgrading LevelDB
+---------------------
+
+Extra care must be taken when upgrading LevelDB. This section explains issues
+you must be aware of.
+
+### File Descriptor Counts
+
```
Comment (sickpig) [inline doc/developer-notes.md:514]: this seems to be the only relevant part of the changes made to `doc/developer-notes.md`, the other one about "subtree" "Scripted diffs", "RPC interface guidelines" and "Git and GitHub tips" should be included on a separate PR (if needed)

**Review by sickpig — APPROVED** [review]
Approved modulo the changes to `doc/developer-notes.md`

File: doc/developer-notes.md [hunk 2]
```diff
@@ -470,3 +470,251 @@ GUI
   - *Rationale*: Model classes pass through events and data from the core, they
     should not interact with the user. That's where View classes come in. The converse also
     holds: try to not directly access core data structures from Views.
+
+Subtrees
+----------
+
+Several parts of the repository are subtrees of software maintained elsewhere.
+
+Some of these are maintained by active developers of Bitcoin Core, in which case changes should probably go
```
Comment (gandrewstone) [inline doc/developer-notes.md]: branding

File: doc/developer-notes.md [hunk 3]
```diff
@@ -470,3 +470,251 @@ GUI
...
+directly upstream without being PRed directly against the project.  They will be merged back in the next
+subtree merge.
+
+Others are external projects without a tight relationship with our project.  Changes to these should also
+be sent upstream but bugfixes may also be prudent to PR against Bitcoin Core so that they can be integrated
```
Comment (gandrewstone) [inline doc/developer-notes.md]: branding

File: doc/developer-notes.md (hunk 1 above)
Comment (gandrewstone) [reply doc/developer-notes.md:514]: I'm ok with the extra stuff in this PR since its a cherry-pick.  But let's fix the branding...

Comment (ptschip) [reply doc/developer-notes.md:514]: ok , fixed the branding.
